src/main/webapp/script.js

```diff
@@ -79,26 +76,89 @@ function initMap() {
       return;
     }
 
-    const curNameElement = document.getElementById('current-name');
-    const curLocElement = document.getElementById('current-loc');
-    curNameElement.innerText = place.name;
-    curLocElement.innerText = place.geometry.location.toString();
-
-    var url = new URL('/places-list', window.location.origin),
+    const url = new URL('/places-list', window.location.origin),
         params = {
           lat: place.geometry.location.lat(),
           lng: place.geometry.location.lng(),
           filter: getDieteryRestrictions()
         }
     Object.keys(params).forEach(key => url.searchParams.append(key, params[key]))
+
     fetch(url).then((response => {
       response.json().then((data) => {
-        getRandomRestaurant(data.results);
+        updateRestaurantList(data.results);
       })
     }))
   });
 }
 
+let selectedElement;
+let selectedIndex;
+
+function updateRestaurantList(restaurants) {
+  console.log(restaurants);
+  let resListElement = document.getElementById("restaurant-list");
+  resListElement.innerHTML = '';
+  for(let index in restaurants) {
+    const { geometry: {location}, placeId, name, photos, rating } = restaurants[index];
+    const photoReference = photos[0].photoReference;
+
+    const resElement = document.createElement('li');
+
+    const nameElement = document.createElement('h3');
+    nameElement.innerHTML = name;
+    resElement.appendChild(nameElement);
+    resElement.id = placeId;
+
+    // const ratingElement = document.createElement('p');
+    // ratingElement.innerHTML = rating;
+    // resElement.appendChild(ratingElement);
+
+    if(photoReference) {
+      const imgElement = document.createElement('img');
+      imgElement.src = "https://maps.googleapis.com/maps/api/place/photo?maxwidth=150&photoreference="
+          + photoReference + "&key=AIzaSyCN-bDjEE_TtP8Md5Mm7RIdhpc6v9h3QoQ";
+      resElement.appendChild(imgElement);
+    }
+
+    resListElement.appendChild(resElement);
+
+    resElement.addEventListener('click', function() {
+      map.panTo(location);
+      map.setZoom(18);
+
+      selectedElement && selectedElement.classList.remove("selected");
+      this.classList.add("selected");
+      selectedElement = this;
+
+      selectedIndex && markers[selectedIndex].setIcon("http://maps.google.com/mapfiles/ms/icons/blue-dot.png");
```

Comment:
Is this to reset the previous selected restaurant?

Reply:
Yup!